cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/ArtifactInspector.java

```diff
@@ -387,6 +407,31 @@ private String getPluginDescription(Class<?> cls) {
     return annotation == null ? "" : annotation.value();
   }
 
+  /**
+   * Returns the metadata mutation for this plugin, return {@code null} if no metadata annotation is there
+   */
+  @Nullable
+  private MetadataMutation getMetadataMutation(PluginId pluginId, Class<?> cls) {
+    Metadata annotation = cls.getAnnotation(Metadata.class);
+    if (annotation == null) {
+      return null;
+    }
+    MetadataScope metadataScope =
+      pluginId.getNamespaceId().equals(NamespaceId.SYSTEM) ? MetadataScope.SYSTEM : MetadataScope.USER;
```

Comment:
Annotations are static information tied directly to the code, it seems like this should always be system scoped metadata and not modifiable by the user.